tests/Sqs/SqsClientTest.php

```diff
@@ -51,6 +51,117 @@ public function testValidatesMd5OfBody()
+            'Body' => 'Test',
+            'MD5OfMessageAttributes' => 'foo',
+            'MessageAttributes' => [
+                'Reference' => [
+                    'BinaryValue' => 'ID-1234',
+                    'DataType' => 'Binary'
+                ],
+                'Name' => [
+                    'StringValue' => 'Bob',
+                    'DataType' => 'String'
+                ],
+                'LastName' => [
+                    'StringValue' => 'Smith',
+                    'DataType' => 'String.LN'
+                ],
+                'Id' => [
+                    'StringValue' => '3.14',
+                    'DataType' => 'Number'
+                ],
+                'Test' => [
+                    'StringValue' => '短发',
+                    'DataType' => 'String'
+                ],
+                'Test2' => [
+                    'StringValue' => 'true',
+                    'DataType' => 'String.短发'
+                ],
+                'name' => [
+                    'StringValue' => 'bob',
+                    'DataType' => 'String'
+                ],
+                'Named' => [
+                    'StringValue' => 'true',
+                    'DataType' => 'String'
+                ],
+            ]
+        ]]]);
+        $this->addMockResults($client, [$mock]);
+        $client->receiveMessage([
+            'QueueUrl' => 'http://foo.com',
+            'MessageAttributeNames' => [
+                'All'
+            ],
+        ]);
+    }
+
+    /**
+     * @expectedException \Aws\Sqs\Exception\SqsException
+     * @expectedExceptionMessage No Attribute MD5 found. Expected 0408bb33aa149494a6a4683d58a7133f
+     */
+    public function testValidatesMd5OfMessageAttributesExists()
+    {
+        $client = new SqsClient([
+            'region'  => 'us-west-2',
+            'version' => 'latest'
+        ]);
+
+        $mock = new Result(['Messages' => [[
+            'Body' => 'Test',
+            'MessageAttributes' => [
+                'Name' => [
+                    'StringValue' => 'Bob',
+                    'DataType' => 'String'
+                ]
+            ]
+        ]]]);
+        $this->addMockResults($client, [$mock]);
+        $client->receiveMessage([
+            'QueueUrl' => 'http://foo.com',
+            'MessageAttributeNames' => [
+                'All'
+            ],
+        ]);
+    }
+
+    /**
+     * @expectedException \Aws\Sqs\Exception\SqsException
+     * @expectedExceptionMessage Attribute MD5 mismatch. Expected foo, found No Attributes
+     */
+    public function testValidatesMessageAttributesExistWithMd5()
+    {
+        $client = new SqsClient([
+            'region'  => 'us-west-2',
+            'version' => 'latest'
+        ]);
+
+        $mock = new Result(['Messages' => [[
+            'Body' => 'Test',
+            'MD5OfMessageAttributes' => 'foo',
+        ]]]);
+        $this->addMockResults($client, [$mock]);
+        $client->receiveMessage([
```

Comment:
I read the thread as 'Now that `SendMessage` allows you to set `MessageAttributes`, can we validate those on `ReceiveMessage` as well?'. @bakura10, your input here would be beneficial if possible.

Validating on `SendMessage{Batch}` could easily be a separate enhancement request if it is wanted, as the format and closure structure will need to change pretty significantly to capture the necessary data.